### DISCUSSION TITLE
Support to remove/update hook

### DISCUSSION THREAD

**Discussion by royprotim** [redis/go-redis#1827] [Q&A]
There is only one method exported on hooks that is to Add. 

Our requirement is to have a new hook for every request. After a redis connection is made, We can only call AddHook as below.

```go
if ds.Redis != nil {
		ds.Redis.AddHook(&QueryLogger{
			Logger: logger,
		})
	}
```

This causes new hooks to append to the list of older ones. So, the it creates log from older hooks as well. 

<img width="796" alt="Screenshot 2021-07-19 at 8 37 48 PM" src="https://user-images.githubusercontent.com/18380085/126182725-22ab47a3-8f84-458e-8d81-23988d309925.png">

Is there any way we can limit the number of hook to one but still be able to update it for each request.

**Comment by vmihailenco** [comment]
You could create a hook that allows swapping hook body. So the hook is added once and then you swap hook body as needed.

**Reply by royprotim** [reply]
As per my understanding, once the hook is added, we will not have access to it from `redis.Client`
